recipe_scrapers/nihhealthyeating.py

```diff
@@ -135,3 +188,28 @@ def recipe_source(self):
         return normalize_string(
             self.soup.find("div", {"id": "Recipe_Source"}).text.split(": ")[1].strip()
         )
+
+    def recipe_cards(self):
+        recipe_cards_maker = self.soup.find(
+            "strong", string=re.compile(r"Recipe Cards:")
+        )
```

Comment:
Nitpick: it isn't necessary to use a regular expression here: this can be simplified to:

```suggestion
recipe_cards_maker = self.soup.find("strong", string="Recipe Cards:")
```

Reply:
Sorry for the delay. Thanks for all the tips! I think I have everything addressed in the latest commit. I think the only thing it seems we're not 100% set on is the dataclass declaration.

I think `purpose` works as well. I updated the latest commit to be clearer about the usage:

```python
@dataclass
class IngredientGroup:
    ingredients: List[str]
    purpose: str = (
        ""  # this group of ingredients is {purpose} (e.g. "For the dressing")
    )
```

Reply:
I also had to set the default value of `purpose` to `""` instead of None. I could not find a way to add mixed types or conditional variables in a dataclass like how we originally declared.

I'm more accustomed to TypeScript where I could do something like `purpose?: string` at the declaration. If there's a proper way to incorporate `None`, I'd be happy to make the switch.

Reply:
Thanks @jedwards1230 - I _think_ the way to do that with Python + `mypy` is to use `Optional` typing.

So the line would change to: `purpose: Optional[str] = None`

(and ensure that the import is added - `from types import List, Optional`, or similar)

Reply:
That seems correct. Thank you! Just pushed.